### DISCUSSION TITLE
"float32-filterable" feature support?

### DISCUSSION THREAD

**Discussion by HymnsForDisco** [gfx-rs/wgpu#4846] [Q&A]
The WebGPU spec has an API for [optional features](https://www.w3.org/TR/webgpu/#features). Likewise, wgpu has a [Features ](https://docs.rs/wgpu/latest/wgpu/struct.Features.html) API, where the docs say 
>  ... Corresponds to [WebGPU GPUFeatureName](https://gpuweb.github.io/gpuweb/#enumdef-gpufeaturename).

I'm a bit confused because the set of "Features" enumerated seems to be missing some (like "float32-filterable"), and contains other "extra" features that aren't mentioned in the WebGPU spec (like SHADER_F64).

Basically, I'm just wondering what's going on here, and if there's any way to enable the "float32-filterable" optional feature with wgpu.

**Comment by kpreid** [comment]
FWIW, I wrote the “Corresponds to…” notes in the docs, and I just meant “this type serves the same role”, for cross-reference, not “it should be expected to be completely identical”.

As to the actual differences, I'd expect but don't know for sure  that they are not so much intentional as `wgpu` not yet supporting all optional functionality defined by WebGPU — and in the other direction, supporting some of its own “extensions”; `wgpu` has quite a few additional things that aren't specified by WebGPU.

**Reply by HymnsForDisco** [reply]
Understood, and thanks for the prompt reply. With that in mind, I'll create an issue / feature request about this.
